### DISCUSSION TITLE
Best way to set SDLC env variables today?

### DISCUSSION THREAD

**Discussion by m-torin** [vercel/vercel#4626] [Help] [answered]
Most posts on this topic are outdated, and those that aren’t gloss over a hard requirement for different env variables depending on a ticket’s swimlane status (like QA, Release x, Staging). For example: https://github.com/vercel/vercel/discussions/4558

The consensus from Vercel seems to be to use the dashboard, but that doesn’t appear to be something that can accommodate anything other than prod, dev and a generic preview. How can we define more stages? Auth0 for example requires clearly defined source and callback domains. 

Thank you for any advice!!!!!

**Comment by styfle** [comment, marked as answer]
There are only 3 environments as you mentioned: Production, Preview, and Development.

However, we might add one or two more in the future.

I think your Auth0 use case can be solved by using the `host` header as the callback in your serverless function, rather than assigning environment variables.

https://vercel.com/docs/v2/edge-network/headers
